### PR TITLE
bluetooth: controller: fixing issue with white list clear

### PR DESCRIPTION
**Pull request by erbr-ot**
When clearing the whitelist, possible white listings in resolve list
were left. These are now also cleared. 

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

### CODE REVIEW

**Comment by cvinayak** [comment]
@joerchan @carlescufi this goes into my backlog for now, if you are unable to review, let me know I will need to pull it to top of my stack.

File: subsys/bluetooth/controller/ll_sw/ull_filter.c [hunk 1]
```diff
@@ -843,6 +843,11 @@ static void wl_clear(void)
 	for (int i = 0; i < WL_SIZE; i++) {
 		wl[i].taken = 0U;
 	}
+
+	for (int i = 0; i < CONFIG_BT_CTLR_RL_SIZE; i++) {
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_filter.c]: Delete this new code. Instead add `rl[j].wl = 0U;` after line 895.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_filter.c (hunk 1 above)
Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_filter.c]: At this point j >= ARRAY_SIZE(rl)?
How come you do not want to remove all 'whitelist refs' in the resolve list when, clearing whitelist?

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_filter.c]: You are right, my fault.

Use lines 908-913 at line 844. This way the for-loop is not needed.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_filter.c (hunk 1 above)
Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_filter.c]: I would actually prefer to keep the for-loop, as it does the same thing, is less code and 'easier to read'.

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_filter.c]: You are adding a new loop with a different iteration count, and wl_clear being in the ll_init call path, introduces Bluetooth initialization time penalty.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_filter.c (hunk 1 above)
Comment (erbr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_filter.c]: OK, I considered this of lesser complexity, but I guess that is not always true. I applied your modification.

**Review by cvinayak — APPROVED** [review]

**Review by wopu-ot — APPROVED** [review]

**Pull request merged**